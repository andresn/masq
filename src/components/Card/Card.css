.Card {
  border-radius: 3px;
  background-color: white;
  box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.12);
}

.Card .flex {
  display: flex;
}

.Card h1 {
  margin: 0;
}

.Card .content {
  padding-left: 32px;
  padding-right: 32px;
  width: 100%;
  margin-top: 16px;
  margin-bottom: 16px;
}

.Card .description {
  color: var(--grey3-color);
  font-size: 14px;
}

.Card .image {
  height: 100px;
  background-size: cover;
  background-position: top center;
  border-radius: 3px;
}
