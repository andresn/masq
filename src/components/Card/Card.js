import React from 'react'
import PropTypes from 'prop-types'

import CardHeader from './CardHeader/CardHeader'

import './Card.css'

export default function Card (props) {
  const { logo, image, title, description, color, enabled, children, onChecked, onTrash } = props
  const styles = {
    title: {
      display: 'flex',
      alignItems: 'center'
    },
    logo: {
      width: '16px',
      height: '16px',
      marginRight: '4px',
      backgroundSize: 'cover',
      backgroundPosition: 'top center',
      backgroundImage: 'url(' + logo + ')'
    }
  }

  const style = enabled
    ? { backgroundImage: 'url(' + image + ')' }
    : {
      backgroundImage: 'url(' + image + ')',
      filter: 'grayscale(50%)'
    }

  return (
    <div className='Card'>
      {image && <div className='image' style={style} />}
      <div className='flex'>
        <div className='content'>
          <CardHeader className='CardHeader' color={color} enabled={enabled} onChecked={onChecked} onTrash={onTrash} />
          {logo
            ? (
              <div style={styles.title}>
                {logo && <div id='logo' style={styles.logo} />}
                <h2>{title}</h2>
              </div>
            )
            : <h2>{title}</h2>
          }
          {description && <p className='description'>{description}</p>}
          {children}
        </div>
      </div>
    </div>
  )
}

Card.propTypes = {
  logo: PropTypes.string,
  image: PropTypes.string,
  children: PropTypes.element,
  description: PropTypes.string,
  color: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  enabled: PropTypes.bool,
  onChecked: PropTypes.func
}
