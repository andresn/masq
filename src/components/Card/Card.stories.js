import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

import Card from './Card'

storiesOf('Card', module)
  .add('enabled', () => (
    <Card enabled color='blue' title='sometitle' />
  ))
  .add('disabled', () => (
    <Card color='blue' title='sometitle' />
  ))
  .add('enabled with a description', () => (
    <Card enabled color='blue' title='sometitle'
      description='some description'
    />
  ))
  .add('enabled with an image', () => (
    <Card enabled color='blue' title='sometitle'
      image='https://images.pexels.com/photos/63703/pexels-photo-63703.jpeg?w=1260&h=750&auto=compress&cs=tinysrgb'
    />
  ))
  .add('enabled with a logo', () => (
    <Card enabled color='blue' title='sometitle'
      logo='https://upload.wikimedia.org/wikipedia/commons/f/fd/Qwant-v3.png'
    />
  ))
  .add('with trash icon', () => (
    <Card
      enabled
      color='blue'
      onTrash={action('onTrash')}
      title='sometitle'
      description='some description'
    />
  ))
