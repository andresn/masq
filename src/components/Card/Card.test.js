import React from 'react'
import Card from './Card'
import renderer from 'react-test-renderer'

it('renders Card disabled', () => {
  const component = renderer.create(<Card enabled color='blue' title='sometitle' />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Card enabled', () => {
  const component = renderer.create(<Card enabled={false} color='blue' title='sometitle' />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Card with a description', () => {
  const component = renderer.create(
    <Card enabled={false} color='blue' title='sometitle'
      description='some description'
    />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Card with an image', () => {
  const component = renderer.create(
    <Card enabled={false} color='blue' title='sometitle'
      image='https://images.pexels.com/photos/63703/pexels-photo-63703.jpeg?w=1260&h=750&auto=compress&cs=tinysrgb'
    />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Card with a logo', () => {
  const component = renderer.create(
    <Card enabled={false} color='blue' title='sometitle'
      logo='https://upload.wikimedia.org/wikipedia/commons/f/fd/Qwant-v3.png'
    />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
