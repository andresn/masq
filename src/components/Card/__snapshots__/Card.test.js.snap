// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card disabled 1`] = `
<div
  className="Card"
>
  <div
    className="flex"
  >
    <div
      className="content"
    >
      <div
        className="CardHeader"
      >
        <div
          className="line"
          style={
            Object {
              "backgroundColor": "blue",
            }
          }
        />
        <div
          className="SwitchButton"
        >
          <label
            style={
              Object {
                "color": "blue",
              }
            }
          >
            ENABLED
          </label>
          <div
            className=""
            id="switch-container"
            onClick={[Function]}
          >
            <div
              className=""
              id="background"
              style={
                Object {
                  "backgroundColor": "blue",
                }
              }
            />
            <div
              className=""
              id="cursor"
              style={
                Object {
                  "transform": "translate3d(26px, 0, 0)",
                }
              }
            />
          </div>
        </div>
      </div>
      <h1>
        sometitle
      </h1>
    </div>
  </div>
</div>
`;

exports[`renders Card enabled 1`] = `
<div
  className="Card"
>
  <div
    className="flex"
  >
    <div
      className="content"
    >
      <div
        className="CardHeader"
      >
        <div
          className="line"
          style={
            Object {
              "backgroundColor": "var(--grey2-color)",
            }
          }
        />
        <div
          className="SwitchButton"
        >
          <label
            style={
              Object {
                "color": null,
              }
            }
          >
            DISABLED
          </label>
          <div
            className=""
            id="switch-container"
            onClick={[Function]}
          >
            <div
              className=""
              id="background"
              style={Object {}}
            />
            <div
              className=""
              id="cursor"
              style={Object {}}
            />
          </div>
        </div>
      </div>
      <h1>
        sometitle
      </h1>
    </div>
  </div>
</div>
`;

exports[`renders Card with a description 1`] = `
<div
  className="Card"
>
  <div
    className="flex"
  >
    <div
      className="content"
    >
      <div
        className="CardHeader"
      >
        <div
          className="line"
          style={
            Object {
              "backgroundColor": "var(--grey2-color)",
            }
          }
        />
        <div
          className="SwitchButton"
        >
          <label
            style={
              Object {
                "color": null,
              }
            }
          >
            DISABLED
          </label>
          <div
            className=""
            id="switch-container"
            onClick={[Function]}
          >
            <div
              className=""
              id="background"
              style={Object {}}
            />
            <div
              className=""
              id="cursor"
              style={Object {}}
            />
          </div>
        </div>
      </div>
      <h1>
        sometitle
      </h1>
      <p
        className="description"
      >
        some description
      </p>
    </div>
  </div>
</div>
`;

exports[`renders Card with a logo 1`] = `
<div
  className="Card"
>
  <div
    className="flex"
  >
    <div
      className="content"
    >
      <div
        className="CardHeader"
      >
        <div
          className="line"
          style={
            Object {
              "backgroundColor": "var(--grey2-color)",
            }
          }
        />
        <div
          className="SwitchButton"
        >
          <label
            style={
              Object {
                "color": null,
              }
            }
          >
            DISABLED
          </label>
          <div
            className=""
            id="switch-container"
            onClick={[Function]}
          >
            <div
              className=""
              id="background"
              style={Object {}}
            />
            <div
              className=""
              id="cursor"
              style={Object {}}
            />
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <div
          id="logo"
          style={
            Object {
              "backgroundImage": "url(https://upload.wikimedia.org/wikipedia/commons/f/fd/Qwant-v3.png)",
              "backgroundPosition": "top center",
              "backgroundSize": "cover",
              "height": "16px",
              "marginRight": "4px",
              "width": "16px",
            }
          }
        />
        <h1>
          sometitle
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Card with an image 1`] = `
<div
  className="Card"
>
  <div
    className="image"
    style={
      Object {
        "backgroundImage": "url(https://images.pexels.com/photos/63703/pexels-photo-63703.jpeg?w=1260&h=750&auto=compress&cs=tinysrgb)",
        "filter": "grayscale(50%)",
      }
    }
  />
  <div
    className="flex"
  >
    <div
      className="content"
    >
      <div
        className="CardHeader"
      >
        <div
          className="line"
          style={
            Object {
              "backgroundColor": "var(--grey2-color)",
            }
          }
        />
        <div
          className="SwitchButton"
        >
          <label
            style={
              Object {
                "color": null,
              }
            }
          >
            DISABLED
          </label>
          <div
            className=""
            id="switch-container"
            onClick={[Function]}
          >
            <div
              className=""
              id="background"
              style={Object {}}
            />
            <div
              className=""
              id="cursor"
              style={Object {}}
            />
          </div>
        </div>
      </div>
      <h1>
        sometitle
      </h1>
    </div>
  </div>
</div>
`;
